src/Enforcer.php

```diff
@@ -472,6 +474,188 @@ public function getImplicitUsersForPermission(string ...$permission): array
         return $res;
     }
 
+    /**
+     * Convert permissions to string as a hash to deduplicate.
+     * 
+     * @param array $permissions
+     * 
+     * @return array
+     */
+    public function removeDumplicatePermissions(array $permissions): array
+    {
+        $permissionsSet = [];
+        $res = [];
+
+        foreach ($permissions as $permission) {
+            $permissionStr = Util::arrayToString($permission);
+
+            if (isset($permissionsSet[$permissionStr])) {
+                continue;
+            }
+
+            $permissionsSet[$permissionStr] = true;
+            $res[] = $permission;
+        }
+        return $res;
+    }
+
+    /**
+     * GetAllowedObjectConditions returns a string array of object conditions that the user can access.
+     * For example: conditions, err := e.GetAllowedObjectConditions("alice", "read", "r.obj.")  
+     * Note:  
+     * 
+     * 0. prefix: You can customize the prefix of the object conditions, and "r.obj." is commonly used as a prefix.
+     * After removing the prefix, the remaining part is the condition of the object.
+     * If there is an obj policy that does not meet the prefix requirement, an ObjConditionException will be thrown.  
+     * 
+     * 1. If the 'objectConditions' array is empty, an EmptyConditionException will be thrown.
+     * This error is thrown because some data adapters' ORM return full table data by default
+     * when they receive an empty condition, which tends to behave contrary to expectations.(e.g. DBALAdapter)
+     * If you are using an adapter that does not behave like this, you can choose to ignore this error.  
+     * 
+     * @param string $user
+     * @param string $action
+     * @param string $prefix
+     * 
+     * @return array
+     * @throws ObjConditionException
+     * @throws EmptyConditionException
+     */
+    public function getAllowedObjectConditions(string $user, string $action, string $prefix): array
+    {
+        $permission = $this->getImplicitPermissionsForUser($user);
+
+        $objectConditions = [];
+        foreach ($permission as $policy) {
+            if ($policy[2] == $action) {
+                if (strpos($policy[1], $prefix) !== 0) {
+                    throw new ObjConditionException('need to meet the prefix required by the object condition');
+                }
+
+                $trimPrefix = function ($string, $prefix) {
+                    if (strpos($string, $prefix) === 0) {
+                        return substr($string, strlen($prefix));
+                    } else {
+                        return $string;
+                    }
+                };
+
+                $objectConditions[] = $trimPrefix($policy[1], $prefix);
```

Comment:
Please simplify this code
```suggestion
                $objectConditions[] = substr($policy[1], strlen($prefix);
```